doc/src/conf.py

```diff
@@ -530,3 +530,21 @@ def linkcode_resolve(domain, info):
 
     fn = os.path.relpath(fn, start=os.path.dirname(sympy.__file__))
     return blobpath + fn + linespec
+
+
+def resolve_type_aliases(app, env, node, contnode):
+    """Resolve :class: references to our type aliases as :attr: instead."""
+    # A sphinx bug means that TypeVar doesn't work:
+    # https://github.com/sphinx-doc/sphinx/issues/10785
+    if (
+        node["refdomain"] == "py"
+        and node["reftype"] == "class"
+        and node["reftarget"] in ["sympy.utilities.decorator.T"]
```

Comment:
I'm okay with the fix. But this looks like we need to add special path to all the typevar references.
I tried building without this extension, but it worked fine though. Maybe you can check the necessity

Reply:
It was failing in CI and also failed locally.

I think I might have found a simpler workaround in gh-27247 though where I used a manual `py:class` reference instead.